pybandits/strategy/multi_objective.py

```
@@ -0,0 +1,607 @@
+        List[np.ndarray]
+            List of Pareto-optimal quantity vectors.
+
+        References
+        ----------
+        The normalized normal constraint method for generating the Pareto frontier (Messac et al., 2003)
+        https://ieeexplore.ieee.org/document/938649
+        """
+        # Step 1: Find anchor points using optimization for each objective
+        anchor_points = [
+            self._objective_selector.verify_and_select_from_quantitative_action(
+                lambda x: func(x)[i], model.models[i], None
```

Ruff (0.15.15)

[warning] Function definition does not bind loop variable `i`

(B023)